packages/gestalt/src/Image.js

```diff
@@ -29,133 +32,109 @@ type Props = {|
 /**
  * https://gestalt.pinterest.systems/Image
  */
-export default class Image extends PureComponent<Props> {
-  static propTypes = {
-    alt: PropTypes.string.isRequired,
-    children: PropTypes.node,
-    color: PropTypes.string,
-    elementTiming: PropTypes.string,
-    fit: PropTypes.oneOf(['contain', 'cover', 'none']),
-    importance: PropTypes.oneOf(['high', 'low', 'auto']),
-    loading: PropTypes.oneOf(['eager', 'lazy', 'auto']),
-    naturalHeight: PropTypes.number.isRequired,
-    naturalWidth: PropTypes.number.isRequired,
-    onError: PropTypes.func,
-    onLoad: PropTypes.func,
-    role: PropTypes.oneOf(['img', 'presentation']),
-    sizes: PropTypes.string,
-    src: PropTypes.string.isRequired,
-    srcSet: PropTypes.string,
-  };
-
-  static defaultProps: {|
-    color: string,
-    fit?: 'contain' | 'cover' | 'none',
-    importance?: 'high' | 'low' | 'auto',
-    loading?: 'eager' | 'lazy' | 'auto',
-  |} = {
-    color: 'transparent',
-    fit: 'none',
-    importance: 'auto',
-    loading: 'auto',
-  };
+export default function Image({
+  alt,
+  children,
+  color = 'transparent',
+  elementTiming,
+  fit = 'none',
+  importance = 'auto',
+  loading = 'auto',
+  naturalHeight,
+  naturalWidth,
+  onError,
+  onLoad,
+  role = 'img',
+  sizes,
+  src,
+  srcSet,
+}: Props): Node {
+  const isScaledImage = shouldScaleImage(fit);
 
-  componentDidMount() {
-    if (shouldScaleImage(this.props.fit)) {
-      this.loadImage();
-    }
-  }
+  const handleLoad = useCallback(() => {
```

Comment:
nit

`  const handleLoad = useCallback(() => onLoad?.(), [onLoad]);
`

Reply:
That actually changes the functionality, since that would now be returning the onLoad call instead of just calling it. That could change behavior if a handler function was returning something that we were previously dropping on the floor, but now we start to return.